src/main/java/emissary/core/channels/LoggingChannelFactory.java

```diff
@@ -0,0 +1,205 @@
+package emissary.core.channels;
+
+import org.apache.commons.lang3.Validate;
+import org.slf4j.Logger;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.SeekableByteChannel;
+import java.util.Arrays;
+import java.util.concurrent.atomic.AtomicLong;
+
+/**
+ * This class creates a SeekableByteChannelFactory for logging accesses to every method on the SeekableByteChannel. It
+ * is intended only for development/debugging purposes.
+ */
+public final class LoggingChannelFactory {
```

Comment:
If this class is meant to be used only during debugging/development, maybe we should consider decorating it with a `@Deprecated` annotation that will generate compiler warnings if any usage gets committed?